a10_neutron_lbaas/handler_base.py

```diff
@@ -13,15 +13,32 @@
 #    under the License.
 
 import json
+import logging
+import threading
+
+LATE_INIT_LOCK = threading.Lock()
+
+LOG = logging.getLogger(__name__)
 
 
 class HandlerBase(object):
 
     def __init__(self, a10_driver):
         self.a10_driver = a10_driver
-        self.hooks = a10_driver.hooks
         self.openstack_driver = self.a10_driver.openstack_driver
 
+        if self.a10_driver.provider is None:
+            LATE_INIT_LOCK.acquire()
```

Comment:
I'm surprised this isn't a context manager, like `with LATE_INIT_LOCK:`. I double checked, and it looks like the interface is acquire ... release.